tests/common/configuration/node_config_model.rs

```diff
@@ -72,25 +77,20 @@ impl NodeConfig {
         }
     }
 
+    pub fn regenerate_ports(&mut self) {
+        self.rest.listen = format!("127.0.0.1:{}", NodeConfig::get_available_port().to_string());
+        self.peer_2_peer.public_address = format!(
+            "/ip4/127.0.0.1/tcp/{}",
+            NodeConfig::get_available_port().to_string()
+        );
+    }
+
     pub fn get_node_address(&self) -> String {
         let output = format!("http://{}/{}", self.rest.listen, self.rest.prefix);
         output
     }
-}
-
-fn get_available_port() -> u16 {
-    let available_port = loop {
-        let port = rand::thread_rng().gen_range(8000, 9999);
-        if port_is_available(port) {
-            break port;
-        }
-    };
-    available_port
-}
 
-fn port_is_available(port: u16) -> bool {
-    match TcpListener::bind(("127.0.0.1", port)) {
-        Ok(_) => true,
-        Err(_) => false,
+    fn get_available_port() -> u16 {
+        NEXT_AVAILABLE_PORT_NUMBER.fetch_add(1, Ordering::SeqCst)
```

Comment:
Much better. Assuming that there is only one test process at a time handing out port numbers. Otherwise there is a running race for ports starting predictably from 8000.

Reply:
Nitpick: it better remain a freestanding function, as there is nothing actually connecting it to `NodeConfig`